omod/src/main/java/org/openmrs/module/pihmalawi/web/taglibs/ETraceAccessTag.java

```diff
@@ -0,0 +1,325 @@
+                    + "\">";
+
+            return link + "Create new " + f.getName() + "</a> on " + dateTag("date-" + uuid, "date-" + uuid);
+        }
+        else {
+
+            link = "<a href=\"/openmrs/module/htmlformentry/htmlFormEntry.form?personId="
+                    + p.getPersonId()
+                    + "&patientId="
+                    + p.getPatientId()
+                    + "&returnUrl=%2fopenmrs%2fpatientDashboard.form&formId="
+                    + f.getFormId() + "\">";
+
+            return link + "Create new " + f.getName() + "</a>";
+        }
+    }
+
+    protected String createEditCardHtmlTag(Patient p, Form f, Encounter initialEncounter) {
+        String link = "";
+        String newMasterCardConfig = getNewMasterCardConfiguration(f);
+        if (newMasterCardConfig != null) {
+            link = "<a href=\"" + newMasterCardConfig + "&patientId="+p.getPatientId()+"\">";
+        }
+        else {
+            link = "<a href=\"/openmrs/module/htmlformentry/htmlFormEntry.form?encounterId=" + initialEncounter.getId() + "&mode=EDIT\">";
+        }
+
+        return link + "Edit " + f.getName() + "</a><br/>" + (includeAppointmentInfo ? getDetails(p, initialEncounter) + "<br/>" : "");
+
+    }
+
+
+
+    protected String getDetails(Patient p, Encounter initialEncounter) {
+        EncounterType followupEncounterType = null;
+        if (StringUtils.isNotBlank(getFollowupEncounterTypeName())) {
+            followupEncounterType = Context.getEncounterService().getEncounterType(getFollowupEncounterTypeName());
+        }
+        if (followupEncounterType == null) {
+            followupEncounterType = Context.getEncounterService().getEncounterType(getFollowupEncounterTypeId());
+        }
+        List<Encounter> followups = Context.getEncounterService().getEncounters(p, null, null, null, null, Arrays.asList(followupEncounterType), null, false);
+        String created = "Created: " + Helper.formatDate(initialEncounter.getEncounterDatetime());
+        String visited = "Visited: no";
+        String rvd = "Appointment: none";
+        if (!followups.isEmpty()) {
+            Encounter lastFollowup = followups.get(followups.size() - 1);
+            visited = "Visited: " + Helper.formatDate(lastFollowup.getEncounterDatetime()) + " at " + lastFollowup.getLocation().getName();
+            Concept appt = Context.getConceptService().getConcept(APPOINTMENT_DATE);
+            Date startOfDay = DateUtil.getStartOfDay(lastFollowup.getEncounterDatetime());
+            Date endOfDay = DateUtil.getEndOfDay(lastFollowup.getEncounterDatetime());
+            List<Obs> os = Context.getObsService().getObservations(Arrays.asList((Person) p), null, Arrays.asList(appt), null, null, null, Arrays.asList("dateCreated"), 1, null, startOfDay, endOfDay, false);
+            if (!os.isEmpty()) {
+                rvd = "Appointment: " + Helper.formatDate(os.get(0).getValueDatetime());
+            }
```

Comment:
Same comments here.  Use a different concept to store next tracking date, change language here to reflect the actual tracking activities.